sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -108,21 +108,20 @@ case class DataSource(
               dataSource
             case Failure(error) =>
               if (error.isInstanceOf[ClassNotFoundException]) {
-                val className = error.getMessage
-                if (spark2RemovedClasses.contains(className)) {
-                  throw new ClassNotFoundException(s"$className is removed in Spark 2.0. " +
+                // error.getMessage is the class name of provider2. Instead, we use provider here.
```

Comment:
In a second thought, I don't think we need this `if` branch. Could you just remove it?